src/QMCDrivers/MCCoords.hpp

```diff
@@ -0,0 +1,129 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_MCCOORDS_HPP
+#define QMCPLUSPLUS_MCCOORDS_HPP
+
+#include "Configuration.h"
+#include "type_traits/complex_help.hpp"
+#include "ParticleBase/RandomSeqGenerator.h"
+
+#include <vector>
+
+namespace qmcplusplus
+{
+
+enum class MCCoordsTypes
+{
+  RS,
+  RSSPINS
+};
+
+template<MCCoordsTypes MCT = MCCoordsTypes::RS>
+struct MCCoords
+{
+  static constexpr MCCoordsTypes mct = MCT;
+  // This cleans up some other code.
+  void resize(const std::size_t size);
+  std::vector<QMCTraits::PosType> rs;
+};
+
+template<>
+struct MCCoords<MCCoordsTypes::RSSPINS>
+{
+  static constexpr MCCoordsTypes mct = MCCoordsTypes::RSSPINS;
+  // This cleans up some other code.
+  void resize(const std::size_t size)
+  {
+    rs.resize(size);
+    spins.resize(size);
+  }
+  std::vector<QMCTraits::PosType> rs;
+  std::vector<std::complex<QMCTraits::RealType>> spins;
+};
+
+template<MCCoordsTypes CT = MCCoordsTypes::RS>
+struct MCCIt
+{
+  std::vector<QMCTraits::PosType>::iterator irs;
+};
+
+template<>
+struct MCCIt<MCCoordsTypes::RSSPINS>
+{
+  std::vector<QMCTraits::PosType>::iterator irs;
+  std::vector<std::complex<QMCTraits::RealType>>::iterator spins;
```

Comment:
maybe use ispins? since you put i in front of rs to indicate iterator, I would be consistent on the spin portion